d2/src/main/java/com/linkedin/d2/discovery/util/D2Utils.java

```diff
@@ -36,7 +36,7 @@ public class D2Utils
   ).collect(Collectors.toSet());
 
   // Keeping the max threshold to 10K, this would ensure that we accidentally won't create more than max ZK tracking nodes.
-  public static final int RAW_D2_CLIENT_MAX_TRACKING_NODE = 10000;
+  public static final int RAW_D2_CLIENT_MAX_TRACKING_NODE = 1000;
```

Comment:
actually the RAW_D2_CLIENT_BASE_PATH could be either /d2/..., or /shared/..., or /kafka/..., varies with the fabric.